### PR TITLE
Fix empty requests

### PR DESCRIPTION
**Pull request by inaod568**
Fix handling of errors occurring when  empty requests are made, when incorrect info is put into Schemas etc. 

Before submitting a pr:
- [x] Tests passing
- [ ] Black formatting
- [ ] Migrations for any changes to the database schema
- [x] Rebase/merge the `dev` branch
- [ ] Note in the CHANGELOG

### CODE REVIEW

File: tests/test_files_new.py [hunk 1]
```diff
@@ -54,6 +54,427 @@ def project_row(project_id):
 # TESTS #################################################################################### TESTS #
 
 
+def test_new_file_empty(client):
```
Comment (talavis) [inline tests/test_files_new.py:57]: Maybe write some kind of helper functions. The code duplication is enormous.

File: tests/test_files_new.py [hunk 2]
```diff
@@ -412,15 +832,14 @@ def test_new_file_nameinbucket_too_short(client):
...
-            data=json.dumps(file_no_nameinbucket),
-            content_type="application/json",
-        )
-
+    response = client.post(
+        tests.DDSEndpoint.FILE_NEW,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client),
+        query_string={"project": "file_testing_project"},
+        data=json.dumps(file_no_nameinbucket),
+        content_type="application/json",
```
Comment (talavis) [inline tests/test_files_new.py:840]:
```suggestion
        json=file_no_nameinbucket,
```

File: tests/test_files_new.py [hunk 3]
```diff
@@ -492,14 +910,14 @@ def test_new_file_publickey_wrong_length(client):
...
-            query_string={"project": "file_testing_project"},
-            data=json.dumps(file_wrong_public_key),
-            content_type="application/json",
-        )
+    response = client.post(
+        tests.DDSEndpoint.FILE_NEW,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client),
+        query_string={"project": "file_testing_project"},
+        data=json.dumps(file_wrong_public_key),
+        content_type="application/json",
```
Comment (talavis) [inline tests/test_files_new.py:918]:
```suggestion
        json=file_wrong_public_key,
```

File: tests/test_files_new.py [hunk 4]
```diff
@@ -432,15 +851,14 @@ def test_new_file_subpath_too_short(client):
...
-            data=json.dumps(file_no_subpath),
-            content_type="application/json",
-        )
-
+    response = client.post(
+        tests.DDSEndpoint.FILE_NEW,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client),
+        query_string={"project": "file_testing_project"},
+        data=json.dumps(file_no_subpath),
+        content_type="application/json",
```
Comment (talavis) [inline tests/test_files_new.py:859]:
```suggestion
        json=file_no_subpath,
```

File: tests/test_files_new.py [hunk 5]
```diff
@@ -510,14 +928,14 @@ def test_new_file_salt_wrong_length(client):
...
-            query_string={"project": "file_testing_project"},
-            data=json.dumps(file_wrong_salt),
-            content_type="application/json",
-        )
+    response = client.post(
+        tests.DDSEndpoint.FILE_NEW,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unitadmin"]).token(client),
+        query_string={"project": "file_testing_project"},
+        data=json.dumps(file_wrong_salt),
+        content_type="application/json",
```
Comment (talavis) [inline tests/test_files_new.py:936]:
```suggestion
        json=file_wrong_salt,
```

**Review by talavis — CHANGES_REQUESTED** [review]
Everything seems to work as intended, but I would strongly advice some improvements in the tests:
* create helper functions for commonly used patterns; there is an enormous amount of code duplication
* use `json=` instead of `data` + `content_type`; I gave some suggestions, but there are many more.

**Comment by inaod568** [comment]
@talavis I agree that there's alot of code duplication and there's in general refactoring to be done, but at this time it's not important to change stuff that's already done, as long as they work as they should. This is a task for the future.

File: dds_web/api/project.py [hunk 6]
```diff
@@ -47,16 +50,20 @@ class ProjectStatus(flask_restful.Resource):
...
 
+        # Get current status and deadline
+        return_info = {"current_status": project.current_status}
         if project.current_deadline:
             return_info["current_deadline"] = project.current_deadline
 
-        if extra_args and extra_args.get("history") == True:
+        # Get status history
+        json_input = flask.request.json
+        if json_input and json_input.get("history") == True:
```
Comment (talavis) [inline dds_web/api/project.py:66]:
```suggestion
        if json_input and json_input.get("history"):
```

**Comment by inaod568** [comment]
Marked as draft, let me know when you find it ok. Remember to not spend heaps of time on it. Code duplication is not a priority. Merging this today

**Review by talavis — APPROVED** [review]

**Pull request merged**